packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-config.ts

```diff
@@ -1,47 +1,93 @@
+import * as cdk from '@aws-cdk/core';
 import { Construct } from 'constructs';
-import { arnForDeploymentConfig } from '../utils';
+import { CfnDeploymentConfig } from '../codedeploy.generated';
+import { ITrafficRoutingConfig } from '../traffic-routing-config';
+import { arnForDeploymentConfig, validateName } from '../utils';
 
 /**
  * The Deployment Configuration of an ECS Deployment Group.
  * The default, pre-defined Configurations are available as constants on the {@link EcsDeploymentConfig} class
  * (for example, `EcsDeploymentConfig.AllAtOnce`).
- *
- * Note: CloudFormation does not currently support creating custom ECS configs outside
- * of using a custom resource. You can import custom deployment config created outside the
- * CDK or via a custom resource with {@link EcsDeploymentConfig#fromEcsDeploymentConfigName}.
+ * To create a custom Deployment Configuration,
+ * instantiate the {@link EcsDeploymentConfig} Construct.
  */
 export interface IEcsDeploymentConfig {
+  /**
+   * @attribute
+   */
   readonly deploymentConfigName: string;
+  /**
+   * @attribute
+   */
   readonly deploymentConfigArn: string;
 }
 
+/**
+ * Construction properties of {@link EcsDeploymentConfig}.
+ */
+export interface EcsDeploymentConfigProps {
+  /**
+   * The physical, human-readable name of the Deployment Configuration.
+   *
+   * @default a name will be auto-generated
+   */
+  readonly deploymentConfigName?: string;
+
+  /**
+   * The configuration that specifies how traffic is shifed from one
+   * Amazon ECS task set to another during an Amazon ECS deployment.
+   *
+   * @default a name will be auto-generated
+   */
+  readonly trafficRoutingConfig?: ITrafficRoutingConfig;
```

Comment:
I preferred the pattern in the ServerDeploymentConfig. I worry about the pattern in `LambdaDeploymentConfig` assumes any potential future types would have a similar shape (`type` + `percentage` + `interval`). Seems if a future deployment config type didn't look like that, then it would be hard to extend the current implementation.

I may be overthinking though 🤷‍♂️